### PR TITLE
init: Throw error when network specific config is ignored

### PR DESCRIPTION
**Pull request by maflcko**
This should have no effect on mainnet users, but simplifies testing, where config settings are currently ignored with only a warning. Fix this by making it an error.

Issues:
*  bitcoin client 0.17.0 ignores wallet's name (file) #14523 
*  Can't set custom rpcport on testnet #13777 
* ...

### CODE REVIEW

**Comment by promag** [comment]
Tested ACK dddd6f0.

```sh
$ cat bitcoin.conf
rpcport=54321

bitcoind -testnet
2019-03-21T14:58:35Z Bitcoin Core version v0.18.99.0-dddd6f0f5 (release build)
2019-03-21T14:58:35Z Error: Config setting for -rpcport only applied on test network when in [test] section.
Error: Config setting for -rpcport only applied on test network when in [test] section.
```

Deserves a small release note, just to point to if someone reports this behavior change?

**Comment by practicalswift** [comment]
Concept ACK

Nice usability improvement!

**Comment by promag** [comment]
@MarcoFalke here's a test for your consideration 53469f780a8e2d88337f4efa6eb9ca52c9ce34f8.

File: doc/release-notes.md [hunk 1]
```diff
@@ -73,9 +73,14 @@ Example item
 Low-level changes
 =================
 
-Example item
+Configuration
 ------------
 
+* An error is issued where previously a warning was issued when a setting in
+  the config file was specified in the default section, but not overriden for
```
Comment (practicalswift) [inline doc/release-notes.md]: Should be "overridden" :-)

**Comment by promag** [comment]
utACK fae38c3.

**Pull request merged**